modules/internal/ChapelBase.chpl

```diff
@@ -1444,7 +1444,10 @@ module ChapelBase {
       // string to a type. Otherwise, we can't resolve chpl_debug_writeln in
       // `range.these`
       { var dummyRange = 1..0; for i in dummyRange {} }
-      return str:t;
+      if t:string == "regex(string)" || t:string == "regex(bytes)" then
```

Comment:
how about `if t == regex(string) || t == regex(bytes)` ?

Reply:
I attempted to make this change and ran into issues. I added some comments to document the problems encountered and noted that we would prefer to use the type comparison rather than string comparison here.